### PR TITLE
Remove duplication in start-php and Pass User Agent in Curl request

### PR DESCRIPTION
**Pull request by smgom**

### CODE REVIEW

**Comment by alovak** [comment]
@smgom looks awesome. Just small styling fixes are required...

File: src/Start/Charge.php [hunk 1]
```diff
@@ -66,14 +38,27 @@ public static function create(array $data)
   */
   public static function all()
   {
-    $url = Start::getEndPoint('charge_list');
-
+       $return_data =  Start_Charge::make_request("charge_list");
```
Comment (alovak) [inline src/Start/Charge.php]: unnecessary spaces after =

File: src/Start/Charge.php [hunk 2]
```diff
@@ -66,14 +38,27 @@ public static function create(array $data)
...
+       return $return_data;
+  }
+  
+  public static function make_request($url, $data =array()){
+    $url = Start::getEndPoint($url);
     $ch = curl_init();
+    if(Start::getUserAgent() != ""){
+        $userAgent = Start::getUserAgent().' / StartPHP ' . Start::VERSION;
+    }else{
+        $userAgent =  'StartPHP ' . Start::VERSION;
```
Comment (alovak) [inline src/Start/Charge.php]: unnecessary spaces after =

File: src/Start/Charge.php [hunk 3]
```diff
@@ -81,7 +66,13 @@ public static function all()
     $errno = curl_errno($ch);
     if( $result === false || $errno != 0 ) {
       // Do error checking
-      throw new Exception(curl_error($ch));
+    if($errno == '1'  || $errno == '35' || $errno == '51' || $errno == '60'){
```
Comment (alovak) [inline src/Start/Charge.php]: extra spaces before ||

File: src/Start/Charge.php [hunk 4]
```diff
@@ -66,14 +38,27 @@ public static function create(array $data)
...
-    $url = Start::getEndPoint('charge_list');
-
+       $return_data =  Start_Charge::make_request("charge_list");
+       return $return_data;
+  }
+  
+  public static function make_request($url, $data =array()){
+    $url = Start::getEndPoint($url);
     $ch = curl_init();
+    if(Start::getUserAgent() != ""){
```
Comment (alovak) [inline src/Start/Charge.php]: as I see in different style guides (we don't have own)

```if(``` should be ```if (```
```}else{``` should be ```} else {```

please update in all files if you find it

**Comment by smgom** [comment]
done.

**Comment by alovak** [comment]
@smgom can you rebase your PR with master? Or merge. Now I can't automatically merge your PR from UI.

**Pull request merged**